dspace-api/src/main/java/org/dspace/rdf/conversion/MetadataRDFMapping.java

```diff
@@ -391,13 +394,30 @@ protected RDFNode parseObject(Model m, RDFNode objectNode, String dsoIRI,
     protected String parseResourceGenerator(Resource resourceGenerator, 
             String value, String dsoIRI)
     {
-        if (resourceGenerator.isURIResource() 
+        if (resourceGenerator.isURIResource()
                 && resourceGenerator.equals(DMRM.DSpaceObjectIRI))
         {
             return dsoIRI;
         }
 
-        return parseValueProcessor(resourceGenerator, value);
+        return getSafeIRI(parseValueProcessor(resourceGenerator, value));
+    }
+
+    private static String getSafeIRI(final String input) {
```

Comment:
Please don't use private here, use protected instead so that inheriting classes can use this method.